module/block_iterator/iterator_test.go

```diff
@@ -32,38 +32,41 @@ func TestIterateHeight(t *testing.T) {
 
 		// create iterator
 		// b0 is the root block, iterate from b1 to b3
-		job := module.IterateRange{Start: b1.Height, End: b3.Height}
+		iterRange := module.IterateRange{Start: b1.Height, End: b3.Height}
 		headers := storagebadger.NewHeaders(&metrics.NoopCollector{}, db)
-		iter, err := NewHeightIterator(headers.BlockIDByHeight, progress, job)
-		require.NoError(t, err)
+		getBlockIDByIndex := func(height uint64) (flow.Identifier, bool, error) {
+			blockID, err := headers.BlockIDByHeight(height)
+			if err != nil {
+				return flow.ZeroID, false, err
+			}
+
+			return blockID, true, nil
+		}
+		iter := NewIndexedBlockIterator(getBlockIDByIndex, progress, iterRange)
 
 		// iterate through all blocks
-		visited := make(map[flow.Identifier]struct{})
+		visited := make([]flow.Identifier, 0, len(bs))
 		for {
 			id, ok, err := iter.Next()
 			require.NoError(t, err)
 			if !ok {
 				break
 			}
 
-			// preventing duplicate visit
-			_, ok = visited[id]
-			require.False(t, ok, fmt.Sprintf("block %v is visited twice", id))
-
-			visited[id] = struct{}{}
+			visited = append(visited, id)
 		}
 
-		// verify all blocks are visited
-		for _, b := range bs {
-			_, ok := visited[b.ID()]
-			require.True(t, ok, fmt.Sprintf("block %v is not visited", b.ID()))
-			delete(visited, b.ID())
+		// verify all blocks are visited in the same order
+		for i, b := range bs {
+			require.Equal(t, b.ID(), visited[i])
 		}
-		require.Empty(t, visited)
 
-		// save the next to iterate height and verify
+		require.Equal(t, len(bs), len(visited))
```

Comment:
do this before the loop to avoid a panic if `visited` is smaller

Reply:
The reason I do the for loop check is so that if there is one block not visited, the test case can tell which block is not visited, rather than saying the total visited is different.